front/app/containers/SearchPage/components/ExportToCsvButton.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import gql from 'graphql-tag';
+import { graphql } from 'react-apollo';
+import * as FontAwesome from 'react-fontawesome';
+import { SiteFragment_siteView } from 'types/SiteFragment';
+import ThemedButton from 'components/StyledComponents/index';
+
+const EXPORT_TO_CSV_MUTATION = gql`
+  mutation ExportToCsvMutation($searchHash: String!, $siteViewId: Int!) {
+    exportToCsv(input: { searchHash: $searchHash, siteViewId: $siteViewId }) {
+      searchExport {
+        id
+      }
+    }
+  }
+`;
+
+interface ExportToCsvButtonProps {
+  siteView: any;
+  searchHash: string;
+  mutate: any;
+  setExportId: any;
+}
+
+class ExportToCsvButton extends React.Component<ExportToCsvButtonProps> {
+  render() {
+    const { mutate, siteView, searchHash, setExportId } = this.props;
+
+    async function onClick() {
+      const { data } = await mutate({
+        variables: { siteViewId: siteView.id, searchHash },
+      });
+      console.log(data);
+      setExportId(data.exportToCsv.searchExport.id);
+    }
+    return (
+      <ThemedButton onClick={onClick}>
+        Export to CSV &nbsp;
+        <FontAwesome name="file-text" />
+      </ThemedButton>
+    );
+  }
+}
+
+// it's a little annoying that the HOC expects so many types
+export default graphql<any, any, any, any>(EXPORT_TO_CSV_MUTATION)(
```

Comment:
Here's an example where TypeScript truly hampers the overall utility of most of React's strategies for minimizing interfaces and getting things done expeditiously. 

I would need to essentially re-declare the props type for the component, but without the mutation, as a separate type, and then also declare the data that we expect to see, the props for the nested component, and the types of all variables. 

See: https://medium.com/@mickeyboston/apollo-typescript-hocs-in-react-explained-93032db3e89a

Reply:
To clarify a bit more, most higher-order components are designed to enrich the nested component with additional props. So the inner component's props is a union of the outer component's props and the props that are added in. Having to explicitly enumerate this takes away almost all the value of the HOC pattern, with little to show for it that you wouldn't get from writing Jest tests anyway. Those are something this project still sorely needs.

Reply:
I agree we need additional tests. However, adding the type information to a HOC isn't actually that painful.  There's a type Omit<T, x> which can take the T of the child component (including the property to be injected) and omit it from the props of the higher order component. 
So the signature for a withX() looks something like this:
`export function withTheme<T>(Component : React.ComponentType<T>) : React.ComponentClass<Omit<T,'theme'>> { ... }`